src/node/interfaces.cpp

```diff
@@ -516,20 +516,26 @@ class ChainImpl : public Chain
     }
     bool haveBlockOnDisk(int height) override
     {
-        LOCK(cs_main);
+        LOCK(::cs_main);
         const CChain& active = Assert(m_node.chainman)->ActiveChain();
         CBlockIndex* block = active[height];
         return block && ((block->nStatus & BLOCK_HAVE_DATA) != 0) && block->nTx > 0;
     }
     CBlockLocator getTipLocator() override
     {
-        LOCK(cs_main);
+        LOCK(::cs_main);
         const CChain& active = Assert(m_node.chainman)->ActiveChain();
         return active.GetLocator();
     }
+    CBlockLocator getActiveChainLocator(const uint256& block_hash) override
+    {
+        LOCK(::cs_main);
+        const CBlockIndex* index = m_node.chainman->m_blockman.LookupBlockIndex(block_hash);
```

Comment:
In commit "wallet: Save wallet scan progress" (6a3088634edaf55dd458d3d5a95d38c868db971a)

I think you need `if (!index) return {};` here so there is a way to detect a block not being found, and to avoid saving the tip as the best block if a block is not found.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25036/commits/86698868cdb878140c7d6a3909d6aa6f024d8cbf. Thanks